plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -42,16 +48,33 @@ public class ThriftMetastoreModule
     @Override
     protected void setup(Binder binder)
     {
-        OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
-                .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
-        binder.bind(TokenAwareMetastoreClientFactory.class).to(StaticTokenAwareMetastoreClientFactory.class).in(Scopes.SINGLETON);
-        configBinder(binder).bindConfig(StaticMetastoreConfig.class);
-        configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
+        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
+        requireNonNull(staticMetastoreConfig.getMetastoreUris(), "metastoreUris is null");
+        boolean hasHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()));
+        boolean hasHttpsMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "https".equalsIgnoreCase(uri.getScheme()));
+
+        if (hasHttpMetastore || hasHttpsMetastore) {
+            OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                    .setDefault().to(HttpThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            binder.bind(IdentityAwareMetastoreClientFactory.class).to(StaticTokenAwareHttpMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            configBinder(binder).bindConfig(StaticMetastoreConfig.class);
+            configBinder(binder).bindConfig(ThriftHttpMetastoreConfig.class);
+            configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
```

Comment:
why do you bind `ThriftMetastoreConfig` here?
It contains a bunch of `hive.metastore.thrift.*` settings.
The idea we discussed about was to have a clean separation between thrift & http related settings.

Reply:
I found that I need to bind `ThriftMetastoreConfig` primarily because the `createWriteStatisticsExecutor` in this class needs it. I could not find a way to avoid loading ThriftHiveConfig due to that. Any ideas? Also, is it goal to not have any http related classes in the thrift path or both ways i.e thrift should not use http and http should not use thrift, I am not convinced why we need to assert the later.

Reply:
I have commented out the line

```
//configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
```
and the trino server still starts successfully.

What is the exception you're getting?